holidays/countries/curacao.py

```diff
@@ -11,81 +11,136 @@
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        super().__init__(*args, **kwargs)
 
     def _populate(self, year):
+        # 1954: Creation of the Netherlands Antilles.
+        if year <= 1953:
+            return None
+
         super()._populate(year)
 
+        # Aña Nobo.
+        # Status: In-Use.
+
         # New Year's Day
-        self[date(year, JAN, 1)] = "Nieuwjaarsdag [New Year's Day]"
+        self._add_new_years_day(tr("Aña Nobo"))
+
+        # Dialuna despues di Carnaval Grandi.
+        # Status: In-Use.
+        # Started in 1947.
+
+        # Carnival Monday
+        self._add_ash_monday(tr("Dialuna despues di Carnaval Grandi"))
 
-        easter_date = easter(year)
-        # Carnaval Monday
-        self[
-            easter_date + td(days=-48)
-        ] = "Maandag na de Grote Karnaval [Carnaval Monday]"
+        # Bièrnèsantu.
+        # Status: In-Use.
 
         # Good Friday
-        self[easter_date + td(days=-2)] = "Goede Vrijdag [Good Friday]"
+        self._add_good_friday(tr("Bièrnèsantu"))
 
-        # Easter Monday
-        self[
-            easter_date + td(days=+1)
-        ] = "Di Dos Dia di Pasku di Resureccion [Easter Monday]"
+        # Pasku di Resurekshon.
+        # Status: In-Use
 
-        # King's Day
-        if year >= 2014:
-            kings_day = date(year, APR, 27)
-            if self._is_sunday(kings_day):
-                kings_day += td(days=-1)
+        # Easter Sunday
+        self._add_easter_sunday(tr("Pasku di Resurekshon"))
 
-            self[kings_day] = "Koningsdag [King's Day]"
+        # Di dos dia di Pasku di Resurekshon.
+        # Status: In-Use.
 
-        # Queen's Day
-        if 1891 <= year <= 2013:
-            queens_day = date(year, APR, 30)
-            if year <= 1948:
-                queens_day = date(year, AUG, 31)
+        # Easter Monday
+        self._add_easter_monday(tr("Di dos dia di Pasku di Resurekshon"))
+
+        # Dia di la Reina/Dia di Rey.
+        # Status: In-Use.
+        # Started under Queen Wilhelmina in 1891.
+        # Queen Beatrix kept Queen Juliana's Birthday after her coronation.
+        # Switched to Aña di Rey in 2014 for King Willem-Alexander.
+        # Have its name changed again to Dia di Rey from 2021 onwards.
+
+        # King's / Queen's Day
+        name = (
+            # King's Day.
+            tr("Dia di Rey")
+            if year >= 2014
+            # Queen's Day.
+            else tr("Dia di la Reina")
+        )
+        if year >= 2014:
+            dt = date(year, APR, 27)
+        else:
+            dt = date(year, APR, 30)
+        if self._is_sunday(dt):
+            dt += td(days=-1) if year >= 1980 else td(days=+1)
+        self._add_holiday(name, dt)
+
+        # Dia di Obrero.
+        # Status: In-Use.
+        # If fall on Sunday, then this will be move to next working day.
+
+        dt = date(year, MAY, 1)
+        if self._is_sunday(dt):
```

Comment:
```suggestion
        if self._is_sunday(dt) or (self._is_monday(dt) and year <= 1979):
```
According to [this](https://loketdigital.gobiernu.cw/Loket/product/571960bbe1e5fe8712b10a1323630e70), if Labor Day falls on a Sunday or on the day when the King's birthday is officially celebrated, then Labor Day will be celebrated on the next succeeding business day. And until 1980, when the Queen's birthday (April 30) fell on a Sunday, it was moved to May 1.